src/Chapter3.hs

```diff
@@ -480,6 +514,49 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data City = City
+  { cityCastle       :: Castle
+  , cityMainBuilding :: Building
+  , cityHouses       :: [Building]
+  }
+
+data Castle
+  = WithoutCastle
+  | JustCastle
+  | SecureCastle
```

Comment:
The task tells that the castle should have a name 🏰  In this case, it can be a `String`. Sorry, if it wasn't clear before. We improved this task wording in the latest version to clarify this moment.
Could you guess, what functions would need patching in case of the type modifications?